src/packet_analysis/protocol/ip/conn_key/vlan_fivetuple/Factory.cc

```diff
@@ -83,9 +90,11 @@ class IPVlanConnKey : public zeek::IPBasedConnKey {
         struct detail::PackedConnTuple tuple;
         // Add 802.1Q vlan tags to connection tuples. The tag representation
         // here is as in the Packet class (where it's oddly 32-bit), since
-        // that's where we learn the tag values from. 0 indicates absence.
+        // that's where we learn the tag values from.
         uint32_t vlan;
         uint32_t inner_vlan;
+        bool vlan_present;
+        bool inner_vlan_present;
```

Comment:
Nice. I guess that makes sense. I wonder if they should be ordered as follows for clarity, but shouldn't really matter.
```
vlan_present
vlan
inner_vlan_present
inner_vlan
```